### PR TITLE
Implementation of argmax and tovec layers

### PR DESCRIPTION
**Pull request by samadejacobs**
Two layers implemented in this PR:

(1) tovec - takes a label/index etc from previous layer and converts to a vector
(2) argmax - argmax of a column vector

### CODE REVIEW

**Comment by timmoon10** [comment]
I've renamed `tovec_layer` to `one_hot_layer` to match [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/one_hot) (I can't find anything equivalent in PyTorch) and added an argmin layer. All of these layers pass unit tests with metric checking. I'm thinking of putting in a GPU implementation before merging them.

**Comment by ndryden** [comment]
> I can't find anything equivalent in PyTorch

PyTorch deliberately avoids one-hot representations, so I don't think you will.

**Comment by timmoon10** [comment]
I've done some work to implement the argmax layer on GPU, but I think it's better to defer that for later. Here's what I have so far: https://github.com/timmoon10/lbann/blob/gpu-argmax-layer/src/layers/misc/argmax.cu.
